ludwig/features/text_feature.py

```diff
@@ -390,27 +390,29 @@ def update_model_definition_with_metadata(
                 for cls in feature_metadata[level_idx2str]
             ]
 
-    @staticmethod
+
     def calculate_overall_stats(
-            test_stats,
-            output_feature,
-            dataset,
+            self,
```

Comment:
@w4nderlust   Please note I changed the text output feature's version of `calculate_overall_stats()` from `@staticmethod` to an instance method.  The reason for this is due to line 401 where I have to access whether the text features is at `word` or `char` level.

Reply:
Any reason `calculate_overall_stats` needs to be static in general?  Do we ever call it without an instance being available?  If it needs to be non-static here, it's probably worth changing all other occurrences to be the same.

Reply:
I think before we were always running it from classes rather than from instances (like we do for preprocessing). See the my other commentf for context.